src/types/composite/array.ts

```diff
@@ -229,12 +233,30 @@ export abstract class BasicArrayType<T extends ArrayLike<unknown>> extends Compo
   }
 
   tree_getValueAtIndex(target: Tree, index: number): T[number] {
+    // a special optimization for uint64
+    if (this.isNumber64UintType) {
```

Comment:
maybe you could make a subclass of BasicArrayType?

A `Number64Array`  that has an elementType of `Number64UintType`.